packages/world/src/modules/core/implementations/BalanceTransferSystem.sol

```diff
@@ -38,6 +39,11 @@ contract BalanceTransferSystem is System, IWorldErrors {
       revert World_InvalidResourceType(RESOURCE_NAMESPACE, toNamespaceId, toNamespaceId.toString());
     }
 
+    // Require the provided namespace name to be empty
+    if (toNamespaceId.getName() != ROOT_NAME) {
+      revert World_InvalidResourceId(toNamespaceId, toNamespaceId.toString());
+    }
```

Comment:
I wonder if we should add an `isNamespace()` helper that checks for type + empty name

Reply:
Yes I'm thinking that; something like `AccessControl` but undecided exactly where it should live

Reply:
I think we could put this in `WorldResourceIdLib`